src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -428,18 +433,99 @@ public void run() {
+    }
+
+    protected int parseHostName(String webAppMessage) throws PluginExecutionException {
+        int httpPrefixIndex = webAppMessage.indexOf(HTTP_PREFIX);
+        if (httpPrefixIndex < 0) {
+            throw new PluginExecutionException("Could not parse the host name from the log message: " + webAppMessage);
+        }
+        int hostNameIndex = httpPrefixIndex + HTTP_PREFIX.length();
+        int portPrefixIndex = webAppMessage.indexOf(":", hostNameIndex);
+        if (portPrefixIndex < 0) {
+            throw new PluginExecutionException("Could not parse the http port number from the log message: " + webAppMessage);
+        }
+        hostName = webAppMessage.substring(hostNameIndex, portPrefixIndex);
+        debug("Parsed host name: " + hostName);
+        return portPrefixIndex;
+    }
+
+    protected void parseHttpPort(String webAppMessage, int portPrefixIndex) {
+        int portIndex = portPrefixIndex + 1;
+        int portEndIndex = webAppMessage.indexOf("/", portIndex);
+        if (portEndIndex < 0) {
+            // if no ending slash, the port ends at the end of the message
+            portEndIndex = webAppMessage.length();
+        }
+        httpPort = webAppMessage.substring(portIndex, portEndIndex);
+        debug("Parsed http port: " + httpPort);
+    }
+
+    protected void parseHttpsPort(List<String> messages) throws PluginExecutionException {
+        for (String message : messages) {
+            debug("Looking for https port in message: " + message);
+            String httpsMessageContents = message.split(LISTENING_ON_PORT_MESSAGE_PREFIX)[1];
+            String[] messageTokens = httpsMessageContents.split(" ");
+            // Look for endpoint with name containing "-ssl"
+            for (String token : messageTokens) {
+                if (token.contains("-ssl")) {
+                    String parsedHttpsPort = getPortFromMessageTokens(messageTokens);
+                    if (parsedHttpsPort != null) {
+                        debug("Parsed https port: " + parsedHttpsPort);
+                        httpsPort = parsedHttpsPort;
+                        return;
+                    } else {
+                        throw new PluginExecutionException("Could not parse the https port number from the log message: " + message);
+                    }
+                }
+            }
+        }
+        debug("Could not find https port. The server might not be configured for https.");
+    }
+
+    private String getPortFromMessageTokens(String[] messageTokens) {
+        String port = null;
+        // For each space-separated token, keep only the numeric parts.
+        // The port is the last numeric token which is a number <= 65535.
+        for (int i = messageTokens.length - 1; i >= 0; i--) {
+            String numericToken = messageTokens[i].replaceAll("[^\\d]", "" );
+            try {
+                int parsedPort = Integer.parseInt(numericToken);
+                if (parsedPort <= 65535) {
```

Comment:
If we check for an upper bound of the port number should we check for a lower bound? ie. 1024

Reply:
There is no lower bound.  User could possibly set the port to one of the well-known ports.